colossalai/inference/config.py

```diff
@@ -393,3 +393,64 @@ class ModelShardInferenceConfig:
     use_cuda_kernel: bool = False
     use_spec_dec: bool = False
     use_flash_attn: bool = False
+
+
+@dataclass
+class GenerationParams:
+    """
+    Param for diffusion
+    """
+
+    # prompt: Union[str, List[str]] = None # NOTE no sure if we should add this param
+    prompt_2: Optional[Union[str, List[str]]] = None
+    prompt_3: Optional[Union[str, List[str]]] = None
+    height: Optional[int] = None
+    width: Optional[int] = None
+    num_inference_steps: int = 28
+    timesteps: List[int] = None
+    guidance_scale: float = 7.0
+    negative_prompt: Optional[
+        Union[str, List[str]]
+    ] = ""  # NOTE(@lry89757) in pixart default to "", in sd3 default to None, we set "" here, but the best solution is change the check_inputs of each forward fuction
+    negative_prompt_2: Optional[Union[str, List[str]]] = None
+    negative_prompt_3: Optional[Union[str, List[str]]] = None
+    num_images_per_prompt: Optional[int] = 1
+    generator: Optional[Union[torch.Generator, List[torch.Generator]]] = None
+    latents: Optional[torch.FloatTensor] = None
+    prompt_embeds: Optional[torch.FloatTensor] = None
+    negative_prompt_embeds: Optional[torch.FloatTensor] = None
+    pooled_prompt_embeds: Optional[torch.FloatTensor] = None
+    negative_pooled_prompt_embeds: Optional[torch.FloatTensor] = None
+    output_type: Optional[str] = "pil"
+    return_dict: bool = True
+    joint_attention_kwargs: Optional[Dict[str, Any]] = None
+    clip_skip: Optional[int] = None
+    callback_on_step_end: Optional[Callable[[int, int, Dict], None]] = None
+    callback_on_step_end_tensor_inputs: List[str] = field(default_factory=lambda: ["latents"])
+
+    def to_dict(self) -> Dict[str, Any]:
+        # NOTE(@lry89757) Only return the dict that not the default value
+        result = {}
+        for field_info in fields(self):
+            name = field_info.name
+            value = getattr(self, name)
+
+            if field_info.default is not MISSING:
+                default = field_info.default
+            else:
+                default = None
+
+            if field_info.default_factory is not MISSING:
```

Comment:
In common cases we should avoid the import and usage of `dataclasses.MISSING`. Refer to https://docs.python.org/3/library/dataclasses.html